pymess/backend/dialer/daktela.py

```diff
@@ -45,25 +47,31 @@ def _update_dialer_states(self, messages):
             resp_message_state = resp_json['result']['statuses'][0]['name'] if len(
                 resp_json['result']['statuses']) else resp_json['result']['action']
             custom_fields = resp_json['result']['customFields']
-            tts_processed = custom_fields['ttsprocessed'][0]
-            whole_message_heard = ('whole_message_heard' in custom_fields and custom_fields['whole_message_heard']
-                                   and custom_fields['whole_message_heard'][0] == 'Yes')
             state_mapped = settings.DIALER_DAKTELA.STATES_MAPPING[resp_message_state]
-            if state_mapped == DialerMessage.STATE.ANSWERED_PARTIAL and whole_message_heard:
-                resp_message_state = str(DialerMessage.STATE.ANSWERED_COMPLETE)
-            if state_mapped == DialerMessage.STATE.DONE and tts_processed == '0':
-                resp_message_state = str(DialerMessage.STATE.NOT_ASSIGNED)
-            message_state = settings.DIALER_DAKTELA.STATES_MAPPING[resp_message_state]
             message_error = resp_json['error'] if len(resp_json['error']) else None
-            tts_processed = resp_json['result']['customFields']['ttsprocessed'][0]
+
+            if message.is_predictive:
+                message_state = state_mapped
+                is_final_state = resp_json['result']['action'] == '5'
+            else:
```

Comment:
this is just reformating of the code and for predictive dialer it checks a result code only  so far. Action detection will be added in the next PR.

I didn't want to refactor this much in this PR (break what works) but I see this can be improved when needed:
- using enums instead of magic numbers
- the robo automatic dialer status check (previous code) is not a general solution and relies heavily on our setup in Daktela admin (customFields) otherwise it fails. The specific code can be moved to the core.
- the request can be made just one or two for each queue instead of one call per record